pkg/ephemeral/ephemeral.go

```diff
@@ -49,37 +66,53 @@ func RegisterSet(set ApplierSet) {
 		Container.Register(set.Container)
 	}
 
+	if set.PersistentVolumeClaim != nil {
+		PersistentVolumeClaim.Register(set.PersistentVolumeClaim)
+	}
+
+	if set.Pod != nil {
+		Pod.Register(set.Pod)
+	}
+
+	if set.Service != nil {
+		Service.Register(set.Service)
+	}
+
 	if set.PodOptions != nil {
 		PodOptions.Register(set.PodOptions)
 	}
 }
 
 // Constraint provides the set of types allowed for appliers and filterers.
 type Constraint interface {
-	kube.PodOptions | corev1.Container
+	kube.PodOptions | corev1.Container | corev1.PersistentVolumeClaim | corev1.Pod | corev1.Service | networkingv1.NetworkPolicy
 }
 
 // Applier is the interface which applies a manipulation to the PodOption to be
 // used to run ephemeral pdos.
 type Applier[T Constraint] interface {
-	Apply(*T)
+	Apply(*T) error
 }
 
 // ApplierFunc is a function which implements the Applier interface and can be
 // used to generically manipulate the PodOptions.
-type ApplierFunc[T Constraint] func(*T)
+type ApplierFunc[T Constraint] func(*T) error
 
-func (f ApplierFunc[T]) Apply(options *T) { f(options) }
+func (f ApplierFunc[T]) Apply(options *T) error { return f(options) }
 
 // ApplierList is an array of registered Appliers which will be applied on
 // a PodOption.
 type ApplierList[T Constraint] []Applier[T]
 
 // Apply calls the Applier::Apply method on all registered appliers.
-func (l ApplierList[T]) Apply(options *T) {
+func (l ApplierList[T]) Apply(options *T) error {
 	for _, applier := range l {
-		applier.Apply(options)
+		if err := applier.Apply(options); err != nil {
+			return err
```

Comment:
I wonder if we should still try to apply the whole set even after the first failure, and collect all the errors into an aggregate error at the end (`errors.Join`). Not sure if there's any point in continuing if one of them fails though 🤷